src/protocol/meson.build

```diff
@@ -0,0 +1,19 @@
+if wayland_scanner.found()
+	prog_wayland_scanner = find_program(wayland_scanner.get_pkgconfig_variable('wayland_scanner'))
+else
+	prog_wayland_scanner = find_program('wayland-scanner')
+endif
+
+gen_client_header = generator(prog_wayland_scanner,
+	output: ['@BASENAME@-client.h'],
+	arguments: ['-c', 'client-header', '@INPUT@', '@BUILD_DIR@/@BASENAME@-client.h'])
+
+gen_private_code = generator(prog_wayland_scanner,
+	output: ['@BASENAME@.c'],
+	arguments: ['-c', 'private-code', '@INPUT@', '@BUILD_DIR@/@BASENAME@.c'])
+
+xdg_client_header = gen_client_header.process('xdg-shell.xml')
+xdg_private_code = gen_private_code.process('xdg-shell.xml')
+
+lsh_client_header = gen_client_header.process('wlr-layer-shell-unstable-v1.xml')
+lsh_private_code = gen_private_code.process('wlr-layer-shell-unstable-v1.xml')
```

Comment:
Can you rename to `xdg_shell_*` and `layer_shell_*`? There are other XDG protocols and lsh is non-obvious.